project/StacksafeScalapbGenerator.scala

```diff
@@ -132,10 +132,16 @@ class StacksafeProtobufGenerator(params: GeneratorParams) extends ProtobufGenera
     super
       .generateSerializedSize(message)(withEqualsAndHashCode)
       .newline
-      .add(
-        "@transient val serializedSizeM = new coop.rchain.models.Memo(coop.rchain.models.ProtoM.serializedSize(this))"
-      )
+      .add("@transient var _serializedSizeM: coop.rchain.models.Memo[Int] = null")
+      .newline
+      .add("def serializedSizeM: coop.rchain.models.Memo[Int] = if(_serializedSizeM == null) {")
```

Comment:
we can either both dig into wikipedia and discuss why this is not a proper double-checked locking implementation and why it may or may not work and why us even having this discussion means that this is hard to maintain...

... or we can use `synchronized` here. Pretty please? :shrek-cat: :)

Reply:
Then the volatile is not needed, I think.

Reply:
forgot :)